### ISSUE TITLE
Razor: SymbolReference locations for @typeparam are misplaced

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#8417]
Based on [this community report](https://community.sonarsource.com/t/error-with-net-8-on-azure-devops-overlapping-symbol-declaration-and-reference-for-symbol-at-range/104762).

### Description

The symbol reference for `@typeparam` symbols are miss-placed when compiled with the .Net8 SDK.

### Repro steps

.Net 8 BlazorWebAssembly
```xml
<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
	<PropertyGroup>
		<ProjectGuid>{1250CC30-A343-4A10-A1DE-B1792ADB6244}</ProjectGuid>
		<TargetFramework>net8.0</TargetFramework>
		<RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
	</PropertyGroup>
	<ItemGroup>
		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="8.0.0" />
		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="8.0.0" PrivateAssets="all" />
		<PackageReference Include="Microsoft.AspNetCore.WebUtilities" Version="8.0.0" />
	</ItemGroup>
</Project>
```

With `TestComponent.razor`
```
@typeparam TSomeVeryLongName where TSomeVeryLongName : class
````

Run the SonarAnalyzsis and generate the `.sonarqube\out\0\output-cs\symrefs.pb` file, like e.g. this one:
[symrefs.zip](https://github.com/SonarSource/sonar-dotnet/files/13488165/symrefs.zip)

```json
{
   "filePath":"C:\\Users\\martin.strecker\\Desktop\\Bench\\Blazor\\TestComponent.razor",
   "reference":[
      {
         "declaration":{
            "startLine":1,
            "endLine":1,
            "startOffset":0,
            "endOffset":17
         },
         "reference":[
            {
               "startLine":1,
               "endLine":1,
               "startOffset":6,
               "endOffset":23
            }
         ]
      }
   ]
}
```

### Expected behavior

The declaration range should be 12-29 and the reference range should be 36-53.

```
@typeparam TSomeVeryLongName where TSomeVeryLongName : class
           ^^^^^^ Decl ^^^^^       ^^^^^ Ref ^^^^^^^
````

### Actual behavior

The declaration range is 0-17 and the reference range is 6-23 and overlaps.
```
@typeparam TSomeVeryLongName where TSomeVeryLongName : class
^^^^^^ Decl ^^^^^
    ^^^^^ Ref ^^^^^^^
````

### Known workarounds

Use .Net 7 SDK (no Symbol references are created).
or
add spaces between `where` and the type parameter reference:
```
@typeparam TSomeVeryLongName where                          TSomeVeryLongName : class
```

**Comment by vedion** [comment]
I can see two pull requests have been merged with master, does that mean this issue is fixed now?

**Comment by martin-strecker-sonarsource** [comment]
We fixed the failure, so the end step is no longer failing. We did so by skipping symbols that overlapped. We keep this ticket open because the root cause still needs to be fixed.

**Comment by pavel-mikula-sonarsource** [comment]
This has been fixed in Roslyn 4.9.2 that was released last week

**Comment by pavel-mikula-sonarsource** [comment]
Ah, probably not yet. While the reference is present, it probably doesn't have the correct offset. @martin-strecker-sonarsource can you try to check?

See #8890 and changes in `SymbolReferenceAnalyzerTest`

**Comment by martin-strecker-sonarsource** [comment]
Yes. The behavior we see in `SymbolReferenceAnalyzerTest` is the overlapping bug. It is under test in the .Net 8 UTs and with the upgrade of Roslyn it is trickling down to the "normal" UTs as well. The comment in the UT describes the wrong behavior we expect to see with the upgrade (we see the (wrong) 6-23 reference that overlaps with it's declaration 0-17):
https://github.com/SonarSource/sonar-dotnet/blob/db66eb7d1e58ec79390decd4d4a500874f9840e5/analyzers/tests/SonarAnalyzer.Test/Rules/Utilities/SymbolReferenceAnalyzerTest.cs#L241-L246

**Comment by martin-strecker-sonarsource** [comment]
I reported the problem in https://github.com/dotnet/razor/issues/10087

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1640](https://sonarsource.atlassian.net/browse/NET-1640)

[NET-1640]: https://sonarsource.atlassian.net/browse/NET-1640?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
